tests/testthat/test-growth_rate.R

```diff
@@ -0,0 +1,56 @@
+library(dplyr)
+
+X <- c(1:10,NA,10:20,NA)
+Y <- c(2^(1:9),NA,NA,2^(10:21))
+
+methods <- c("rel_change","linear_reg","smooth_spline","trend_filter")
+
+gr <- function(method = "rel_change", h = 3, na_rm = TRUE, ...) {
+  growth_rate(x=X,y=Y,method=method,na_rm = na_rm, h = h,...)
+}
+
+test_that("Test error throwing",{
+  # Error cases
+  expect_error(growth_rate(x=1:3,y=1:4),
+               "`x` and `y` must have the same length.")
```

Comment:
suggestion [non-blocking]: Probably beyond scope here, but it would be nice if the errors and warnings in `growth_rate` had classes so we could check for that rather than message wording, which is liable to change.